scarb/src/core/source/id.rs

```diff
@@ -30,6 +30,8 @@ pub struct SourceIdInner {
     pub kind: SourceKind,
     /// The canonical URL of this source, used for internal comparison purposes.
     pub canonical_url: CanonicalUrl,
+    /// For example, the exact Git revision of the specified branch for a Git Source.
+    pub precise: Option<String>,
```

Comment:
Back when initially implementing this struct I was thinking of putting `precise` within `Git` kind because we won't use it elsewhere? Wdyt?

Reply:
Tbh I am ok with both of them :)